### PR TITLE
3.16 has clarified the range of numbers as being 32bit.

### PR DESCRIPTION
**Pull request by kjeremy**
See: https://microsoft.github.io/language-server-protocol/specifications/specification-3-16/#number

### CODE REVIEW

File: src/req_queue.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ pub struct Incoming<I> {
 
 #[derive(Debug)]
 pub struct Outgoing<O> {
-    next_id: u32,
+    next_id: i32,
     pending: HashMap<RequestId, O>,
 }
```
Comment (matklad) [inline src/req_queue.rs:33]: Commit message is wrong?

> Request Ids should be unsigned

should be *signed*?

**Comment by kjeremy** [comment]
@matklad This needs to be merged before we can update rust-analyzer

**Comment by matklad** [comment]
Could you bump major in Cargo.toml as well and rebase?

I think I've setup auto-release here

bors d+

**Comment by matklad** [comment]
also, negative ids are lol. If I were evil, I'd change rust-analyzer to count server requests ids from zero *down*.

**Comment by matklad** [comment]
Seems like auto-publishing works :tada: !

**Pull request merged**